### PR TITLE
Flat list and Tree Exploration

### PR DESCRIPTION
**Pull request by sahil143**
This PR introduces:
- support for flat-list when `showTree` checkbox is `unchecked`
- show non-matching siblings in tree-view

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
@sahil143 disable in-line addition of children in Flat View
https://docs.google.com/presentation/d/19vRm9GTG_3CEB89syBNzUrJvXZeak-KXs_WmpLd7CZE/edit#slide=id.g2c696760bb_1_21 (comments section)

**Comment by vikram-raj** [comment]
Work Item doesn't maintain the order of execution. When I toggle between the flat view and tree view.

For the reference: https://www.youtube.com/watch?v=jmPisPOTf4A

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

File: src/app/components/planner-list/planner-list.component.ts [hunk 1]
```diff
@@ -488,30 +491,28 @@ export class PlannerListComponent implements OnInit, AfterViewChecked, OnDestroy
...
-          ...this.getParentIdsAll(this.workItems),
-          ...this.getParentIdsAll(this.included)
-        ];
-        this.datatableWorkitems = [
-          ...this.tableWorkitem(this.workItems, null, true),
-          ...this.tableWorkitem(this.included, null, false)
-        ];
-        this.workItems = [...this.workItems, ...this.included];
+        
+          this.included = this.workItemService.resolveWorkItems(
```
Comment (jarifibrahim) [inline src/app/components/planner-list/planner-list.component.ts]: Please fix the indentation

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

File: src/app/components/planner-list/planner-list.component.html [hunk 2]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
+            <ngx-datatable-column
+              *ngFor="let c of (columns | filterColumn:'display')"
+              [isTreeColumn]="c.name === 'Title'"
+              [name]="c.name"
+              [resizeable]="false"
+              [width]="c.width"
+              [sortable]="c.name === 'Title'"
+              [cellClass]="'cell-class'"
+              [headerClass]="'header-cell'">
+                <ng-template ngx-datatable-cell-tree-expander *ngIf="c.name==='Title'">
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: Give space before and after ``` === ```. As you did it in line number 208. Do the same to maintain the consistency throughout the code.

File: src/app/components/planner-list/planner-list.component.html [hunk 3]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
+                  <i class="fa fa-angle-right tree-icon"></i>
+                </ng-template>
 
-              <ng-template ngx-datatable-cell-tree-collapser *ngIf="c.name==='Title'">
-                <i class="fa fa-angle-down tree-icon"></i>
-              </ng-template>
+                <ng-template ngx-datatable-cell-tree-collapser *ngIf="c.name==='Title'">
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: Same here space before and after ```===```.

File: src/app/components/planner-list/planner-list.component.html [hunk 4]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
+                  <i class="fa fa-angle-down tree-icon"></i>
+                </ng-template>
 
-              <ng-template ngx-datatable-cell-tree-disabled *ngIf="c.name==='Title'">
-                <i class="fa fa-angle-right tree-icon tree-icon__disabled"></i>
-              </ng-template>
+                <ng-template ngx-datatable-cell-tree-disabled *ngIf="c.name==='Title'">
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: Same here space before and after ```===```.

File: src/app/components/planner-list/planner-list.component.html [hunk 5]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
+                  <i class="fa fa-angle-right tree-icon tree-icon__disabled"></i>
+                </ng-template>
 
-              <ng-template ngx-datatable-cell-tree-loader *ngIf="c.name==='Title'">
-                <div class="spinner spinner-xs icon-loader margin-top-15"></div>
-              </ng-template>
+                <ng-template ngx-datatable-cell-tree-loader *ngIf="c.name==='Title'">
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: Same here space before and after ```===```.

File: src/app/components/planner-list/planner-list.component.html [hunk 6]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
+                <ng-template let-row="row"
+                  ngx-datatable-cell-template>
+                    <work-item-cell
+                      [row]="row"
+                      [col]="c.prop"
+                      (onDetailPreview)="onDetailPreview($event)"
+                      (onQuickPreview)="onPreview($event)"
+                      (clickLabel)="onClickLabel($event)"
+                      [class.f8-wi__table-config]="c.prop=='label' || c.prop=='assignees'"
+                      [class.planner-hack-title-truncate]="c.prop=='title'">
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: ```===``` would be good in place of ```==```.

File: src/app/components/planner-list/planner-list.component.html [hunk 7]
```diff
@@ -199,45 +202,84 @@ <h4>Please wait, we are loading your data.</h4>
...
```
Comment (vikram-raj) [inline src/app/components/planner-list/planner-list.component.html]: ```===``` would be good in place of ```==```.

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-6 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2432-7 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2432-fabric8-planner.badger.fabric8.io

**Pull request merged**